client/lists/infinite-scroll-list.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback, useEffect, useRef } from 'react'
+import styled from 'styled-components'
+import LoadingIndicator from '../progress/dots'
+import { usePropAsRef } from '../state-hooks'
+
+const LoadingArea = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+  min-height: 56px;
+  padding: 16px 0;
+`
+
+interface InfiniteListProps {
+  children: React.ReactNode
+  /** Whether the functionality of loading data at the beginning of the list is enabled. */
+  prevLoadingEnabled?: boolean
+  /** Whether the functionality of loading data at the ending of the list is enabled. */
+  nextLoadingEnabled?: boolean
+  /** Whether we are currently loading data at the beginning of the list. */
+  isLoadingPrev?: boolean
+  /** Whether we are currently loading data at the ending of the list. */
+  isLoadingNext?: boolean
+  /** Whether this list has data that could be loaded at the beginning of the list. */
+  hasPrevData?: boolean
+  /** Whether this list has data that could be loaded at the ending of the list. */
+  hasNextData?: boolean
+  /**
+   * A unique string value which will restart the intersection observer, i.e. disconnect and start
+   * observing again, when it changes. This is useful when the same instance of this component is
+   * used for lists with different content, e.g. in tabs, chat channels, and whisper sessions.
+   */
+  refreshToken?: string
```

Comment:
You're not passing this as a string in some places already (e.g. carousel), so I don't think this type is right? There's no need for it to be a string either, anything that can be compared is fine, so I'd just make this `unknown`